### DISCUSSION TITLE
Use generated secret for `ImageRepository`

### DISCUSSION THREAD

**Discussion by loxy** [fluxcd/flux2#2947] [Q&A] [answered]
Hi!

Is it somehow possible to use a generated secret (`secretGenerator` in an Kustomize file) for the `secretRef` in an `ImageRepository`? Seems not to work because of the added suffix. Or how do you setup such a private `ImageRepository`?

Best regards,
Kersten

**Comment by makkes** [comment, marked as answer]
[`disableNameSuffixHash: true`](https://kubernetes.io/docs/tasks/manage-kubernetes-objects/kustomization/#generatoroptions) should work.

**Reply by loxy** [reply]
Yes, I know. That's what I use currently. But then you loose this automatic behavior when content of secret is changing...

**Reply by hiddeco** [reply]
What automatic behavior? If the content of the Secret changes, the controller will just look at whatever the contents are at the next interval of reconciling the `ImageRepository`.

**Reply by makkes** [reply]
If the content changes and the Secret name stays the same, Flux (i.e. kustomize-controller) will update the Secret object in the cluster.
